src/Compiler/Symbols/FSharpDiagnostic.fs

```diff
@@ -67,6 +67,17 @@ module ExtendedData =
     [<Interface; Experimental("This FCS API is experimental and subject to change.")>]
     type IFSharpDiagnosticExtendedData = interface end
 
+    /// Additional data for diagnostics about obsolete attributes.
+    [<Class; Experimental("This FCS API is experimental and subject to change.")>]
+    type ObsoleteDiagnosticExtendedData
```

Comment:
@auduchinok Would `DiagnosticId, UrlFormat` properties be enough for Rider to support something like this in F#.

<img width="879" alt="Screenshot 2025-01-10 at 11 59 06" src="https://github.com/user-attachments/assets/b9bc757b-3ed4-4d38-8ea1-dfd56072fe9c" />